### PR TITLE
[version update] Upgraded dependencies and added requires.io badge #55

### LINKED ISSUE #55: [requirements] Upgrade dependencies and add requires.io badge
**Reserved for GCI**

- check the outdated dependencies on https://requires.io/github/openwisp/django-ipam/requirements/
- upgrade the dependencies in the requirement files
- ensure all tests pass, if there's any failure please fix it
- add the same requirements badge from requires.io as we have in the other openwisp modules
- send a pull request

### PR DESCRIPTION
**Pull request by EdgeKing810**
Updated requirements.txt so that it uses the latest versions of python modules. Also, added the badge of django-ipam from requires.io. I'll squash the commits if everything is fine.

GCI task: https://codein.withgoogle.com/dashboard/task-instances/6728753267867648/

Closes #55

### CODE REVIEW

**Comment by EdgeKing810** [comment]
Been constantly trying to fix that travis build... I'll work on it later today.

**Comment by EdgeKing810** [comment]
Update: I don't know if I did it right, but it works!

**Review by atb00ker — COMMENTED** [review]
Hi, 

Please follow the commit guidelines more closely, it's very important at the time of publishing the application to the end-user and a clean history is very important because we will forget the work of the commit in the coming years (or maybe months!) 😀

Your commit message is 
```
Updated the requirements with the latest version of modules.

Update README.rst

Updated with requires.io badge.

Updated .travis.yml with latest Django Version

Flake8 issues due to pycodestyle version.

Updated requirements-test.txt -> Flake8 issues due to pycodestyle version.

Reverted back the changes.

Trying to fix travis build.
```

Can you please convert it to something more understandable in future, like 
```
Upgrade dependencies and add requires.io badge

Closes #55
```

Remember you can always use `git commit --amend` to amend your commit. 

Thanks!

File: django_ipam/base/fields.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ def to_python(self, value):
 
         try:
             network = ip_network(value, strict=False)
-        except ValueError as e:
+        except ValueError:
```
Comment (atb00ker) [inline django_ipam/base/fields.py:30]: Also, why was this changed in this commit, doesn't look like it's related to the issue, any particular reason? :smile:

Comment (nemesifier) [reply django_ipam/base/fields.py:30]: maybe because of a `flake8` warning since that `e` doesn't seem to be used, so I think it's ok

File: requirements.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
-django>=1.11,<2.1
+django>=1.11,<2.2
 swapper>=1.1.0
 openwisp-utils>=0.2,<0.3
-djangorestframework>=3.7,<3.9
+djangorestframework>=3.7,<4.0
```
Comment (nemesifier) [inline requirements.txt]: change to `3.10`, we don't know if they're doing to release the `4.0`

**Review by nemesifier — COMMENTED** [review]
@EdgeKing810 good progress, see my comments for minor changes

**Comment by nemesifier** [comment]
PS: do also what @atb00ker wrote about the commit message please, remember to use `git commit --amend`

File: requirements.txt (hunk 2 above)
Comment (EdgeKing810) [reply requirements.txt]: Hey... what should I change to "3.10"? I am not understading...

Comment (EdgeKing810) [reply requirements.txt]: You meant "4.10" right?

**Review by EdgeKing810 — COMMENTED** [review]
Changed djangorestframework max version to 4.10.

**Comment by EdgeKing810** [comment]
I won't lie but I looooooooooooveee squashing commits. Is everything good now?

Oh and to not comment more (annoys and makes me feel guilty on Gitter), I'm adding this part:
For each squashed commit that I "produce"... I need only the main part, I mean, I don't need to include a minor description of all the small commits I made? I'll start doing that as from now if you tell me.

File: requirements.txt (hunk 2 above)
Comment (atb00ker) [reply requirements.txt]: `3.9` -> `3.10` (Not `4.10`)!

Comment (EdgeKing810) [reply requirements.txt]: Great! I understood I think. I should apply this when I'm developing software too! Thanks for having introduced me to that.

**Review by nemesifier — COMMENTED** [review]
@EdgeKing810 no you don't need to include a small note of all the changes you have done, just a summary.

Only if a commit is big, important or needs to be explained you need to add more information.

**Comment by EdgeKing810** [comment]
Oh okay, I'll do that better from now on.

**Review by nemesifier — APPROVED** [review]
